### PR TITLE
feat: add updateStartAndEndTimes & roundFee

### LINKED ISSUE #1158: Round Implementation explore size reduction
Explore reducing the bytecode size of the RoundImplementation by grouping update RoundTime functions into 1

### PR DESCRIPTION
**Pull request by bhargavaparoksham**
##### Description

- Reduces the bytecode size of the RoundImplementation by grouping update RoundTime functions into 1
- Adds roundFee

##### Refers/Fixes

fixes #1158

### CODE REVIEW

**Comment by melissa-neira** [comment]
defer to engineers 🐙

File: packages/contracts/contracts/round/RoundImplementation.sol
Comment (thelostone-mc) [inline packages/contracts/contracts/round/RoundImplementation.sol:306]: @gravityblast

Comment (gravityblast) [reply packages/contracts/contracts/round/RoundImplementation.sol:306]: LGTM!

**Comment by thelostone-mc** [comment]
Don't merge this until we have the fee + amount

File: packages/contracts/contracts/round/RoundImplementation.sol
Comment (thelostone-mc) [inline packages/contracts/contracts/round/RoundImplementation.sol:315]:
```suggestion
    require(newApplicationsStartTime <= newRoundStartTime, "Round: Round start is before application start");
```

File: packages/contracts/contracts/round/RoundImplementation.sol
Comment (thelostone-mc) [inline packages/contracts/contracts/round/RoundImplementation.sol:316]:
```suggestion
    require(newApplicationsEndTime <= newRoundEndTime, "Round: Round end is before application end");
```

File: packages/contracts/contracts/round/RoundImplementation.sol
Comment (thelostone-mc) [inline packages/contracts/contracts/round/RoundImplementation.sol:318]:
```suggestion
    if(newApplicationsStartTime != applicationsStartTime) {
```

Comment (thelostone-mc) [reply packages/contracts/contracts/round/RoundImplementation.sol:318]: Maybe we shift the block.timestamp check to one level above and do something like

```
require(newApplicationsStartTime >= block.timestamp
```

this check would be the only block.timestamp check needed for the entire function

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**
